### PR TITLE
Add forgot password page

### LINKED ISSUE #132: FRONTEND: Create Forgot Password Page
Build up the Forgot Password page/route based on the Figma design pdf, pages 24 & 43. This page will eventually function with the backend and email.

### PR DESCRIPTION
**Pull request by dbridenbeck**
### What issue is this solving?
Closes #132 

### Any helpful knowledge/context for the reviewer?
Is a `npm install` necessary?: Nope!
Any special requirements to test?: Yes, you need to add this to app.py on the backend to suppress sending any actual emails when submitting the form on the Forgot Password page:

```
app.config['MAIL_SUPPRESS_SEND'] = True
```

Also note that I created a reusable `Modal` component as part of this issue. I tried to make it fairly flexible and un-opinionated so that it could be easily used in other contexts.

### Please make sure you've attempted to meet the following coding standards
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code
- [x] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

If you're having trouble meeting this criteria, feel free to reach out on Slack for help!

### CODE REVIEW

**Comment by NickSchimek** [comment]
@dbridenbeck It may be good to commit this change `app.config['MAIL_SUPPRESS_SEND'] = True` as the default config for now with this PR so that the app doesn't appear broken to other devs who may want to play with this endpoint. Issue #307 will take care of this issue, but until it is done it might be good to set it as true for now.

**Comment by NickSchimek** [comment]
I tested live and this works. There are some improvements that can be done in the backend to improve the UX on the frontend. For example: sending emails in either a background job or asynchronously to eliminate the lag a user sees when they click submit on the form. <- I plan to eventually create a ticket for this.

As far as I can see this implements half of the reset password functionality. I'm not sure if there is another issue to implement the other half. If not, it will need to be created to implement the functionality when a user clicks on the link in the email - they should be taken to a page that allows them to reset their password. (Note: Current URL's set in the backend email template will need to be updated to whatever URL the frontend requires).

We will need to decide wether or not we want to automatically log the user in when resetting their password or not.

File: src/scss/pages/_dashboard.scss [hunk 1]
```diff
@@ -80,26 +80,26 @@
...
-			padding-right: 45px;
-			margin-left: 20px;
-			font-weight: bold;
-			&:first-child {
-				margin-left: 0;
-			}
-		}
-	}
-}
+// .modal {
```
Comment (Chris-Boe) [inline src/scss/pages/_dashboard.scss]: Let's delete these since we aren't using them anymore

File: src/views/dashboard.js [hunk 2]
```diff
@@ -144,7 +144,8 @@ export const Dashboard = (props) => {
                     </div>
                 </div>
             </div>
-            <div className={`modal ${modalActive && 'is-active'}`}>
+            {/* TODO replace this with our modal! Marie and Darren are smart :) */}
```
Comment (Chris-Boe) [inline src/views/dashboard.js]: 😆

**Review by Chris-Boe — CHANGES_REQUESTED** [review]
Looks great! Just needs to be rebased and have the merge conflicts resolved. Let's also add the modal component to the reusable docs if it's ready!

**Comment by dbridenbeck** [comment]
Just pushed up my changes, and also added PropTypes to the Modal component. 

Note that I tried testing this again but upon trying to pull down the backend I ran into issues and am unable to get the backend running anymore :( 

Any chance you could test this in the UI again? I know a lot of work has been done in the backend recently and it might be worth double-triple checking that the forgot password page is still working.

Also, running `git rebase master` when on this feature branch should do the trick for rebasing, right?

**Comment by dbridenbeck** [comment]
Alright, I resolved the merge conflicts and had to do some tinkering with the logic on the dashboard for opening/closing the modal, along with some CSS to ensure that the buttons had some better spacing between each other when hasButtons = true for the Modal component.

I tested this by setting SUPPRESS_MAIL_SEND to 'true' in config.py and all is working well. My main concern here is the heavy amount of merge commits I had to fight through, so another set of eyes on this would be helpful!

**Review by Chris-Boe — APPROVED** [review]
Looks good!

**Pull request merged**